.github/workflows/build.yml

```diff
@@ -32,15 +41,32 @@ jobs:
           const latestDevCommit = "${{ steps.latest_commit.outputs.commit }}"
           const { owner, repo } = context.repo
           let { data: release } = await github.rest.repos.getLatestRelease({ owner, repo })
-          const latestReleaseCommit = release.body.split('@')[1]
+          const firstLine = release.body.split('\n')[0]
+          const latestReleaseCommit = firstLine.split('@')[1]
           console.log(`Latest release commit: ${latestReleaseCommit}`)
           console.log(`Latest ruby commit: ${latestDevCommit}`)
-          if (latestReleaseCommit === latestDevCommit) {
-            return 'false'
-          } else {
-            return 'true'
-          }
-        result-encoding: string
+          core.setOutput('should_build', latestReleaseCommit !== latestDevCommit)
+          core.setOutput('previous_release', release.tag_name)
+    - name: Compute build and reuse matrix
+      uses: actions/github-script@v7
+      id: matrix
+      with:
+        script: |
+          const osList = ['ubuntu-20.04', 'ubuntu-22.04', 'ubuntu-24.04', 'ubuntu-22.04-arm', 'ubuntu-24.04-arm', 'macos-13', 'macos-14']
+          const skipSlow = "${{ github.event_name == 'workflow_dispatch' && github.event.inputs.skip_slow == 'true' }}"
+          const buildMatrix = JSON.stringify({
+            os: osList,
+            name: skipSlow === 'false' ? ['head', 'debug'] : ['head'],
+            ...(skipSlow === 'false' && { include: [{ os: 'ubuntu-24.04', name: 'asan' }] })
```

Comment:
This seems a bit too magic for my JS taste, could you just do a `if` below and mutate the object?

Reply:
Simplified in ae399e4d97e3f621eeb9a114002737fbc5328e92